Makefile

```
@@ -18,6 +18,7 @@ help:
 	@echo "  format         run blackdoc, docformatter and ruff to automatically format the code"
 	@echo "  check          run code style and quality checks (blackdoc, docformatter, ruff)"
 	@echo "  codespell      run codespell to check common misspellings"
+	@echo "  typecheck      run Mypy for static type check"
```

Looking at https://mypy.readthedocs.io/en/stable/index.html, it doesn't seem like they capitalize `mypy` in the middle of a sentence.
```suggestion
	@echo "  typecheck      run mypy for static type check"
```